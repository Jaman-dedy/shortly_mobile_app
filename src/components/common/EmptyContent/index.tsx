import React from 'react';
import {s} from 'react-native-size-matters';

import EmptyContentImg from '@app/assets/images/empty.svg';
import {Box, Text} from '@app/assets/style/theme';

const EmptyContent = () => {
  return (
    <Box padding="sm">
      <Text variant="paragraph">Sorry, your history is empty!</Text>
      <Box alignItems="center">
        <EmptyContentImg height={s(280)} width={250} />
      </Box>
      <Text variant="paragraph">
        click on the button below to shorten more links
      </Text>
    </Box>
  );
};

export default EmptyContent;
