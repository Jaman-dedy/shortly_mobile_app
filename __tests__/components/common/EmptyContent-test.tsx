import React from 'react';
import renderer from 'react-test-renderer';
import {cleanup} from '@testing-library/react-native';

import EmptyContent from '../../../src/components/common/EmptyContent';

import MockThemes from '../../../mock/MockTheme';

afterEach(cleanup);
test('renders correctly', async () => {
  const tree = await renderer
    .create(
      <MockThemes>
        <EmptyContent />
      </MockThemes>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
