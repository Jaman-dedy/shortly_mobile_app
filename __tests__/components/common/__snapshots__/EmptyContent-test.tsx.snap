// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "padding": 8,
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#35323E",
          "fontFamily": "Poppins-Medium",
          "fontSize": 18,
          "textAlign": "center",
        },
      ]
    }
  >
    Sorry, your history is empty!
  </Text>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
        },
      ]
    }
  >
    <
      height={600}
      width={250}
    />
  </View>
  <Text
    style={
      Array [
        Object {
          "color": "#35323E",
          "fontFamily": "Poppins-Medium",
          "fontSize": 18,
          "textAlign": "center",
        },
      ]
    }
  >
    click on the button below to shorten more links
  </Text>
</View>
`;
